### PR TITLE
pd: add a missing metric in async requests

### PR DESCRIPTION
**Pull request by overvenus**
Add a missing metric in async requests.

Note: schedule time will be recorded too.

@siddontang @BusyJay

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

CI failed

File: src/pd/async/util.rs [hunk 1]
```diff
@@ -231,18 +230,28 @@ impl<Req, Resp, F> Request<Req, Resp, F>
...
-                    Ok(resp) => {
-                        self.resp = Some(Ok(resp));
-                        Ok(self)
-                    }
-                    Err(err) => {
-                        error!("request failed: {:?}", err);
-                        Err(self)
+
+        ok(self)
+            .and_then(|mut ctx| {
```
Comment (andelf) [inline src/pd/async/util.rs]: is map better?

**Comment by siddontang** [comment]
PTAL @andelf

File: src/pd/async/util.rs (hunk 1 above)
Comment (overvenus) [inline src/pd/async/util.rs]: Addressed in commit https://github.com/pingcap/tikv/pull/1974/commits/d8fb52cc037192493adc74791ac6c37e41b0796b and https://github.com/pingcap/tikv/pull/1974/commits/40227014d19563bfd1fddd77d32592c73ff67dcc

**Comment by siddontang** [comment]
PTAL @disksing

**Comment by disksing** [comment]
LGTM.

**Pull request merged**